<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="./">
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Workflow configuration reference &mdash; IPANEMAP Suite  documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=92fd9be5" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css?v=19f00094" />

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="_static/jquery.js?v=5d32c60e"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="_static/documentation_options.js?v=5929fcd5"></script>
        <script src="_static/doctools.js?v=9a2dae69"></script>
        <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Folder structure" href="folders.html" />
    <link rel="prev" title="Data import" href="stepbystep.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            IPANEMAP Suite
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Getting started</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="index.html">Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="requirements.html">Requirements</a></li>
<li class="toctree-l1"><a class="reference internal" href="install.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="run.html">Run the pipeline</a></li>
<li class="toctree-l1"><a class="reference internal" href="configuration.html">Configuration</a></li>
<li class="toctree-l1"><a class="reference internal" href="output.html">Output and Reports</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Go further</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="stepbystep.html">Data import</a></li>
<li class="toctree-l1"><a class="reference internal" href="stepbystep.html#data-conversion">Data conversion</a></li>
<li class="toctree-l1"><a class="reference internal" href="stepbystep.html#qushape-project-generation">QuShape project generation</a></li>
<li class="toctree-l1"><a class="reference internal" href="stepbystep.html#qushape-treatment-of-shape-ce-data">QuShape treatment of SHAPE-CE data</a></li>
<li class="toctree-l1"><a class="reference internal" href="stepbystep.html#qushape-reactivity-extraction">QuShape reactivity extraction</a></li>
<li class="toctree-l1"><a class="reference internal" href="stepbystep.html#reactivity-normalization">Reactivity Normalization</a></li>
<li class="toctree-l1"><a class="reference internal" href="stepbystep.html#reactivity-aggregation">Reactivity Aggregation</a></li>
<li class="toctree-l1"><a class="reference internal" href="stepbystep.html#structure-generation-ipanemap">Structure generation - IPANEMAP</a></li>
<li class="toctree-l1"><a class="reference internal" href="stepbystep.html#footprinting">Footprinting</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Workflow configuration reference</a></li>
<li class="toctree-l1"><a class="reference internal" href="folders.html">Folder structure</a></li>
<li class="toctree-l1"><a class="reference internal" href="troubleshooting.html">Troubleshooting</a></li>
<li class="toctree-l1"><a class="reference internal" href="modules.html">Pipeline scripts references</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">IPANEMAP Suite</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Workflow configuration reference</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/config_yaml_ref.md.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="workflow-configuration-reference">
<h1>Workflow configuration reference<a class="headerlink" href="#workflow-configuration-reference" title="Link to this heading"></a></h1>
<dl>
<dt>samples</dt><dd><p>path to the samples.tsv file that contains
one row per sample. It can be parsed easily via pandas and opened by
libreoffice and excel
Typically:</p>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">samples</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;config/samples.tsv&quot;</span>
</pre></div>
</div>
</dd>
<dt>sequences</dt><dd><p>List of studied RNA, with an identifier, and the corresponding fasta file
The fasta file should contain a description line containing the origin of the sequence (sequence database) with its unique identifier
all “rna_id” column value in samples.tsv must have a corresponding line in this section.</p>
<p><em>Example:</em></p>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">sequences</span><span class="p">:</span>
<span class="w">  </span><span class="nt">sequence_id_1</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;resources/sequence1.fa&quot;</span>
<span class="w">  </span><span class="nt">sequence_id_2</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;resources/sequence2.fa&quot;</span>
<span class="w">  </span><span class="nt">sequence_id_3</span><span class="p">:</span><span class="w"> </span><span class="s">&quot;resources/sequence3.fa&quot;</span>
</pre></div>
</div>
</dd>
<dt>conditions</dt><dd><p>List each condition here, each item in this line must correspond to a column of the samples.tsv file, and be also putted into format -&gt; condition
<em>Example:</em></p>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">conditions</span><span class="p">:</span>
<span class="w">  </span><span class="p p-Indicator">-</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">probe</span>
<span class="w">  </span><span class="p p-Indicator">-</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">temperature</span>
<span class="w">  </span><span class="p p-Indicator">-</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">magnesium</span>
<span class="w">  </span><span class="p p-Indicator">-</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">...</span>
</pre></div>
</div>
</dd>
<dt>allow_auto_import</dt><dd><p>Allow snakemake to import automatically external files
Not fully tested yet</p>
</dd>
<dt>format</dt><dd><dl>
<p>Section to define filename and titles formatting</p>
<dt>condition</dt><dd><p>Format of the conditions, will be use to identify sample and construct filename.
All variable conditions should be referenced here
Name of the conditions must correspond a column header in the samples.tsv file</p>
<p><em>Example:</em></p>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">format</span><span class="p">:</span>
<span class="w">  </span><span class="l l-Scalar l-Scalar-Plain">...</span>
<span class="w">  </span><span class="l l-Scalar l-Scalar-Plain">condition</span><span class="p p-Indicator">:</span><span class="w"> </span><span class="s">&quot;{probe}_{temperature}_{magnesium}&quot;</span>
<span class="w">  </span><span class="l l-Scalar l-Scalar-Plain">...</span>
</pre></div>
</div>
</dd>
<dt>control_condition</dt><dd><p>Same as condition but for control data. {control} will be replaced by the rawdata-&gt;control variable content</p>
<p><em>Example:</em></p>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">format</span><span class="p">:</span>
<span class="w">  </span><span class="l l-Scalar l-Scalar-Plain">...</span>
<span class="w">  </span><span class="l l-Scalar l-Scalar-Plain">control_condition</span><span class="p p-Indicator">:</span><span class="w"> </span><span class="s">&quot;{control}_of_{probe}_{temperature}C_{magnesium}&quot;</span>
<span class="w">  </span><span class="l l-Scalar l-Scalar-Plain">...</span>
</pre></div>
</div>
</dd>
<dt>message</dt><dd><p>Format use to create comprehensible Snakemake message when running workflow. conditions must be prefixed with “wildcards.”</p>
<p><em>Example:</em></p>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">format</span><span class="p">:</span>
<span class="w">  </span><span class="l l-Scalar l-Scalar-Plain">...</span>
<span class="w">  </span><span class="l l-Scalar l-Scalar-Plain">message</span><span class="p p-Indicator">:</span><span class="w"> </span><span class="s">&quot;{wildcards.rna_id}</span><span class="nv"> </span><span class="s">with</span><span class="nv"> </span><span class="s">{wildcards.probe}</span><span class="nv"> </span><span class="s">with</span><span class="nv"> </span><span class="s">{wildcards.temperature}°C</span><span class="nv"> </span><span class="s">{wildcards.magnesium}&quot;</span>
<span class="w">  </span><span class="l l-Scalar l-Scalar-Plain">...</span>
</pre></div>
</div>
</dd>
</dl>
</dd>
<dt>rawdata</dt><dd><dl>
<p>Information concerning raw datas</p>
<dt>path_prefix</dt><dd><p>Path to the folder containing Raw experimental data, it will be prefixed to “control_file” and “probe_file” to auto import data from this folder. Absolute path, and relative path from project root are accepted.</p>
<p>*Example:</p>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">rawdata</span><span class="p">:</span>
<span class="w">  </span><span class="l l-Scalar l-Scalar-Plain">...</span>
<span class="w">  </span><span class="l l-Scalar l-Scalar-Plain">path_prefix</span><span class="p p-Indicator">:</span><span class="w"> </span><span class="s">&quot;path/to/my/raw/data/folder</span>
<span class="w">  </span><span class="s">...</span>
</pre></div>
</div>
</dd>
<dt>type</dt><dd><p>Type of the raw data. It will establish the starting point of the pipeline. Either from file directly from the sequencer or already converted to csv.
Accepted values : “fluo-ceq8000” or “fluo-ce”</p>
<p><em>Example:</em></p>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">rawdata</span><span class="p">:</span>
<span class="w">  </span><span class="l l-Scalar l-Scalar-Plain">...</span>
<span class="w">  </span><span class="l l-Scalar l-Scalar-Plain">type</span><span class="p p-Indicator">:</span><span class="w"> </span><span class="s">&quot;fluo-ceq8000&quot;</span>
<span class="w">  </span><span class="l l-Scalar l-Scalar-Plain">...</span>
</pre></div>
</div>
</dd>
<dt>control</dt><dd><p>Name of the control</p>
<p><em>Example:</em></p>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">rawdata</span><span class="p">:</span>
<span class="w">  </span><span class="l l-Scalar l-Scalar-Plain">...</span>
<span class="w">  </span><span class="l l-Scalar l-Scalar-Plain">control</span><span class="p p-Indicator">:</span><span class="w"> </span><span class="s">&quot;DMSO&quot;</span>
<span class="w">  </span><span class="l l-Scalar l-Scalar-Plain">...</span>
</pre></div>
</div>
</dd>
</dl>
</dd>
<dt>qushape</dt><dd><p>QuShape project generator &amp; extractor
This section allow to configure the way snakemake will generate QuShape
project. those informations must be identical to the configuration of your
capillary sequencer</p>
<p><code class="docutils literal notranslate"><span class="pre">channel</span></code> section corresponds to the definition of sequencer channels
used for this project, here, channel start at index 0: channel 1
correspond to value 0
<code class="docutils literal notranslate"><span class="pre">use_subsequence</span></code> boolean tells if want the pipeline to manage RNA longer that what
your sequencer can manage, in which case, you use multiple reverse transcription
primer, in order to have several starting point for your sequencing. And you want the
workflow to assemble those subsequence into a full RNA.
<code class="docutils literal notranslate"><span class="pre">run_qushape</span></code> you can ask the workflow to launch qushape while running. you
must also fill in <code class="docutils literal notranslate"><span class="pre">run_qushape_conda_env</span></code> with the conda env containing
qushape
<code class="docutils literal notranslate"><span class="pre">check_integrity</span></code> true by default, and highly recommanded to stay true. With true, the workflow will check if the sequence inside the
QuShape File is the same as the sequence declared in sample.tsv and config.yaml. If
you use QuShape file not generated by the workflow, you might have QuShape File with a
another sequence (Longer for example). In which case the test will fail. Only if you
know what your are doing you might want to disable this test. Warning: wrong sequences can
lead to incoherent data, duplicated entry in output file and other inconsistancy.
<em>Example:</em></p>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">qushape</span><span class="p">:</span>
<span class="w"> </span><span class="nt">check_integrity</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">true</span>
<span class="w"> </span><span class="nt">use_subsequence</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">false</span>
<span class="w"> </span><span class="nt">run_qushape</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">false</span>
<span class="w"> </span><span class="nt">run_qushape_conda_env</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">qushape</span>
<span class="w"> </span><span class="nt">channels</span><span class="p">:</span>
<span class="w">   </span><span class="nt">RX</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0</span><span class="w"> </span><span class="c1"># Channel 1</span>
<span class="w">   </span><span class="nt">RXS1</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">2</span><span class="w"> </span><span class="c1"># Channel 3</span>
<span class="w">   </span><span class="nt">BG</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0</span><span class="w"> </span><span class="c1"># Channel 1</span>
<span class="w">   </span><span class="nt">BGS1</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">2</span><span class="w"> </span><span class="c1"># Channel 3</span>
</pre></div>
</div>
</dd>
<dt>normalization</dt><dd><p>Reactivity Normalization configuration</p>
<p>items in this sections are equivalent of the flags available when running</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>python<span class="w"> </span>workflow/scripts/tools/normalize_reactivity.py<span class="w"> </span>--help
</pre></div>
</div>
<p>you can refer to this for more accurate information</p>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">normalization</span><span class="p">:</span>
<span class="w">  </span><span class="c1"># Which nucleotide are reactive to the shape probe</span>
<span class="w">  </span><span class="nt">reactive_nucleotides</span><span class="p">:</span><span class="w"> </span><span class="p p-Indicator">[</span><span class="s">&quot;A&quot;</span><span class="p p-Indicator">,</span><span class="w"> </span><span class="s">&quot;C&quot;</span><span class="p p-Indicator">,</span><span class="w"> </span><span class="s">&quot;G&quot;</span><span class="p p-Indicator">,</span><span class="w"> </span><span class="s">&quot;U&quot;</span><span class="p p-Indicator">]</span>
<span class="w">  </span><span class="c1"># All value above this percentile are considered as outliers, preliminarly to all other treatments.</span>
<span class="w">  </span><span class="nt">stop_percentile</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">90.</span>
<span class="w">  </span><span class="c1"># Value below this threshold are considered non-reactive (0.0)</span>
<span class="w">  </span><span class="nt">low_norm_reactivity_threshold</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">-0.3</span>
<span class="w">  </span><span class="c1"># Which normalizations methods will be used, several are accepted. Autorized values are : simple, interquartile</span>
<span class="w">  </span><span class="nt">norm_methods</span><span class="p">:</span>
<span class="w">    </span><span class="p p-Indicator">-</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">simple</span>
<span class="w">    </span><span class="p p-Indicator">-</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">interquartile</span>
<span class="w">  </span><span class="c1"># oulier threshold for simple normalization method</span>
<span class="w">  </span><span class="nt">simple_outlier_percentile</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">98.</span>
<span class="w">  </span><span class="c1"># percentile of values averaged to create the normalization term</span>
<span class="w">  </span><span class="nt">simple_norm_term_avg_percentile</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">90.</span>
</pre></div>
</div>
</dd>
<dt>aggregate</dt><dd><p>Replicate Aggregation configuration</p>
<p>items in this sections are equivalent of the flags available when running</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>python<span class="w"> </span>workflow/scripts/tools/aggregate_reactivity.py<span class="w"> </span>--help
</pre></div>
</div>
<p>you can refer to this for more accurate information</p>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">aggregate</span><span class="p">:</span>
<span class="w">  </span><span class="nt">norm_method</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">simple</span><span class="w"> </span><span class="c1"># or interquartile</span>

<span class="w">  </span><span class="c1"># You can specify the name of the column used for normalization instead of the normalization method</span>
<span class="w">  </span><span class="c1">#norm_column: &quot;simple_norm_reactivity&quot; #</span>
<span class="w">  </span><span class="nt">min_ndata_perc</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0.5</span>
<span class="w">  </span><span class="nt">min_nsubdata_perc</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0.66</span>
<span class="w">  </span><span class="nt">max_mean_perc</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0.682</span>
<span class="w">  </span><span class="nt">min_dispersion</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">0.05</span>
</pre></div>
</div>
</dd>
<dt>ipanemap</dt><dd><p>IPANEMAP configuration</p>
<p>in order to calculate structures, you can define in which way you want to
combine conditions as input for ipanemap. the <code class="docutils literal notranslate"><span class="pre">pool</span> <span class="pre">section</span></code> enable you to do
so.</p>
<p><em>Example:</em></p>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">ipanemap</span><span class="p">:</span>
<span class="w">  </span><span class="l l-Scalar l-Scalar-Plain">...</span>
<span class="w">  </span><span class="l l-Scalar l-Scalar-Plain">pools</span><span class="p p-Indicator">:</span>
<span class="w">    </span><span class="p p-Indicator">-</span><span class="w"> </span><span class="nt">id</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">1M7_noMg_37C</span>
<span class="w">      </span><span class="nt">rna_id</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">sequence1</span>
<span class="w">      </span><span class="nt">conditions</span><span class="p">:</span>
<span class="w">        </span><span class="p p-Indicator">-</span><span class="w"> </span><span class="nt">temperature</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">37</span>
<span class="w">          </span><span class="nt">magnesium</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">noMg</span>
<span class="w">          </span><span class="nt">probe</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">1M7</span>
<span class="w">    </span><span class="p p-Indicator">-</span><span class="w"> </span><span class="nt">id</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">allProbe_noMg_37C</span>
<span class="w">      </span><span class="nt">rna_id</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">sequence1</span>
<span class="w">      </span><span class="nt">conditions</span><span class="p">:</span>
<span class="w">        </span><span class="p p-Indicator">-</span><span class="w"> </span><span class="nt">temperature</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">37</span>
<span class="w">          </span><span class="nt">magnesium</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">noMg</span>
<span class="w">          </span><span class="nt">probe</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">1M7</span>
<span class="w">        </span><span class="p p-Indicator">-</span><span class="w"> </span><span class="nt">temperature</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">37</span>
<span class="w">          </span><span class="nt">magnesium</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">noMg</span>
<span class="w">          </span><span class="nt">probe</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">NMIA</span>
<span class="w">        </span><span class="p p-Indicator">-</span><span class="w"> </span><span class="nt">temperature</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">37</span>
<span class="w">          </span><span class="nt">magnesium</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">noMg</span>
<span class="w">          </span><span class="nt">probe</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">BzCN</span>
<span class="w">  </span><span class="l l-Scalar l-Scalar-Plain">...</span>
</pre></div>
</div>
<p>each run of IPANEMAP can be entirerly configure.
to create un new run (or <code class="docutils literal notranslate"><span class="pre">pool</span></code>) add a dash in the <code class="docutils literal notranslate"><span class="pre">pools</span></code> sections
each pool should have an unique <code class="docutils literal notranslate"><span class="pre">id</span></code> of your choice.</p>
<p><code class="docutils literal notranslate"><span class="pre">rna_id</span></code> correspond to the sequence to input to IPANEMAP this id must
correspond to and id of the <code class="docutils literal notranslate"><span class="pre">sequences</span></code> section of this configuration file.</p>
<p><code class="docutils literal notranslate"><span class="pre">conditions</span></code> correspond of the sets of conditions you want to be added to
IPANEMAP run. each set of condition start with a dash and must contains all
conditions references in the <code class="docutils literal notranslate"><span class="pre">conditions</span></code> section of this configuration file.</p>
<p><em>In the Example:</em></p>
<ul class="simple">
<li><p>first pool (1M7_noMg_37C) will be executed with “sequence1”
and with one condition:</p>
<ul>
<li><p>all sample aggregated which have no magnesium, a temperature of 37C,
and 1M7 probing</p></li>
</ul>
</li>
<li><p>second pool (allProbe_noMg_37C) will be executed with “sequence1”
and with 3 conditions :</p>
<ul>
<li><p>all sample aggregated which have no magnesium, a temperature of 37C, and
1M7 probing</p></li>
<li><p>all sample aggregated which have no magnesium, a temperature of 37C, and
NMIA probing</p></li>
<li><p>all sample aggregated which have no magnesium, a temperature of 37C, and
BzCN probing</p></li>
</ul>
</li>
</ul>
<p>Parameters in sampling, clustering, pareto, visualization are equivalent to
the content of ipanemap config file. Please refer to IPANEMAP documentation</p>
<p><a class="reference external" href="https://github.com/afafbioinfo/IPANEMAP">https://github.com/afafbioinfo/IPANEMAP</a></p>
</dd>
<dt>folders</dt><dd><p>Folder names</p>
<p>Intermediary folder naming is fully customable,</p>
<p><em>Default values:</em></p>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">folders</span><span class="p">:</span>
<span class="w">  </span><span class="nt">fluo-ceq8000</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">1.1-fluo-ceq8000</span>
<span class="w">  </span><span class="nt">fluo-ce</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">1.2-fluo-ce</span>
<span class="w">  </span><span class="nt">qushape</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">2-qushape</span>
<span class="w">  </span><span class="nt">reactivity</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">3.1-reactivity</span>
<span class="w">  </span><span class="nt">normreact</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">3.2-normreact</span>
<span class="w">  </span><span class="nt">aggreact</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">4.1-aggreact</span>
<span class="w">  </span><span class="nt">aggreact-ipanemap</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">4.2-aggreact-ipanemap</span>
<span class="w">  </span><span class="nt">ipanemap-config</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">5.1-ipanemap-config</span>
<span class="w">  </span><span class="nt">ipanemap-out</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">5.2-ipanemap-out</span>
<span class="w">  </span><span class="nt">structure</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">5.3-structure</span>
<span class="w">  </span><span class="nt">varna</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">5.4-varna</span>
</pre></div>
</div>
</dd>
</dl>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="stepbystep.html" class="btn btn-neutral float-left" title="Data import" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="folders.html" class="btn btn-neutral float-right" title="Folder structure" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024, François-Xavier Lyonnet du Moutier, Nan Pan, Sebastian Will.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>